test/peripheral_test.dart

```diff
@@ -0,0 +1,407 @@
+
+      //then
+      expect(characteristicWithValue, mockedCharacteristicWithValue);
+    });
+
+    test("should use autogenerated transactionId during writing operation", () async {
+      //given
+      var ids = ["4", "9"];
+      when(transactionIdGeneratorMock.getNextId()).thenAnswer((_) => ids.removeAt(0));
+
+      //when
+      await peripheral.writeCharacteristic('s1', 'c1', Uint8List.fromList([1]), false);
+      await peripheral.writeCharacteristic('s1', 'c1', Uint8List.fromList([1]), false);
+
+      //then
+      verify(managerForPeripheral.writeCharacteristicForDevice(any, any, any, any, any, '4'));
+      verify(managerForPeripheral.writeCharacteristicForDevice(any, any, any, any, any, '9'));
+    });
+
+    test("should return values of monitored characteristic", () {
+      //given
+      var emittedValues = [
+        CharacteristicWithValueMock(),
+        CharacteristicWithValueMock(),
+        CharacteristicWithValueMock()
+      ];
+      when(managerForPeripheral.monitorCharacteristicForDevice(any, SERVICE_UUID, CHARACTERISTIC_UUID, TRANSACTION_ID))
+          .thenAnswer((_) => Stream.fromIterable(emittedValues));
+
+      //when
+      Stream<CharacteristicWithValue> characteristicsStream =  peripheral.monitorCharacteristic(SERVICE_UUID, CHARACTERISTIC_UUID, transactionId: TRANSACTION_ID);
+
+      //then
+      expect(characteristicsStream, emitsInOrder(emittedValues));
+    });
+
+    test("should use autogenerated transactionId for monitoring operation", () {
+      //given
+      var ids = ["4", "9"];
+      when(transactionIdGeneratorMock.getNextId()).thenAnswer((_) => ids.removeAt(0));
+
+      //when
+     peripheral.monitorCharacteristic('s1', 'c1');
+     peripheral.monitorCharacteristic('s1', 'c1');
+
+      //then
+      verify(managerForPeripheral.monitorCharacteristicForDevice(any, any, any, '4'));
+      verify(managerForPeripheral.monitorCharacteristicForDevice(any, any, any, '9'));
+    });
+
+  });
+
+  group("Rssi", () {
+    test("use given transactionId", () async {
+      //given
+      const transactionId = "86791384-";
+      const rssi = 23;
+      when(managerForPeripheral.rssi(any, transactionId)).thenAnswer((_) => Future.value(rssi));
+
+      //when
+      int obtainedRssi = await peripheral.rssi(transactionId: transactionId);
+
+      //then
+      verify(managerForPeripheral.rssi(any, transactionId));
+      expect(obtainedRssi, rssi);
+    });
+
+    test("use generated transactionId", () async {
+      //given
+      var ids = ["4", "9"];
+      when(transactionIdGeneratorMock.getNextId()).thenAnswer((_) => ids.removeAt(0));
+
+      //when
+      await peripheral.rssi();
+      await peripheral.rssi();
+
+      //then
+      verify(managerForPeripheral.rssi(any, '4'));
+      verify(managerForPeripheral.rssi(any, '9'));
+    });
+  });
+
+  group("MTU", () {
+    test("returns negotiated MTU", () async {
+      //given
+      const requestedMtu = 23;
+      const negotiatedMtu = 135;
```

Comment:
These two values would be reversed, ie. lower would be the one negotiated, in the wild. Perhaps we should set it this way in the test in a case someone new to BLE decides to reach to them for help?